src/api/block.js

```diff
@@ -1,20 +1,41 @@
 'use strict'
 
-const argCommand = require('../cmd-helpers').argCommand
+const promisify = require('promisify-es6')
 
 module.exports = (send) => {
   return {
-    get: argCommand(send, 'block/get'),
-    stat: argCommand(send, 'block/stat'),
-    put (file, cb) {
+    get: promisify((args, opts, callback) => {
+      if (typeof (opts) === 'function') {
+        callback = opts
+        opts = {}
+      }
+      return send({
+        path: 'block/get',
+        args: args,
+        qs: opts
+      }, callback)
+    }),
+    stat: promisify((args, opts, callback) => {
+      if (typeof (opts) === 'function') {
+        callback = opts
+        opts = {}
+      }
```

Comment:
We do the function signature check/switch in a lot of places. Perhaps we should put that in a function?